src/Parser/Core/Modules/Items/FlaskChecker.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import SUGGESTION_IMPORTANCE from 'Parser/Core/ISSUE_IMPORTANCE';
+import SPELLS from 'common/SPELLS';
+
+const MAX_FLASK_IDS = [
+  SPELLS.FLASK_OF_THE_CURRENTS.id,
+  SPELLS.FLASK_OF_ENDLESS_FATHOMS.id,
+  SPELLS.FLASK_OF_THE_UNDERTOW.id,
+  SPELLS.FLASK_OF_THE_VAST_HORIZON.id,
+];
+
+class FlaskChecker extends Analyzer {
+
+  startFightWithFlaskUp = false;
+
+  on_toPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if(MAX_FLASK_IDS.includes(spellId) && event.prepull) {
+      this.startFightWithFlaskUp = true;
+    }
+  }
+
+  get flaskSuggestionThresholds() {
+    return {
+      actual: this.startFightWithFlaskUp,
+      isEqual: false,
+      style: 'boolean',
+    };
+  }
+
+  suggestions(when) {
+    when(this.flaskSuggestionThresholds)
+      .addSuggestion((suggest) => {
+        return suggest(<React.Fragment>You did not had a flask up before combat.  Having a flask during the combat increase your primary stat by a fair amount.</React.Fragment>)
```

Comment:
iirc since you don't actually have any react elements within the text, you can just use a string. (i.e. `"..."` instead of `<React.Fragment>...</React.Fragment>`